### DISCUSSION TITLE
HSM

### DISCUSSION THREAD

**Discussion by Babacar98** [Keyfactor/ejbca-ce#722] [Q&A]
I have deployed EJBCA and SignServer in the cloud to allow my clients to sign via APIs. However, the cost of HSMs on AWS is high. Can this service be replaced by AWS KMS? What alternatives or cost-reduction strategies do you recommend for managing digital signatures securely without exceeding my budget?

**Comment by primetomas** [comment]
[AWS CloudHSM](https://docs.keyfactor.com/ejbca/latest/aws-cloudhsm) is special in it's handling of PKCS#11 and is supported in EJBCA Enterprise.
[AWS KMS](https://docs.keyfactor.com/ejbca/latest/aws-kms) is REST based and also supported in EJBCA Enterprise.

**Comment by Babacar98** [comment]
thanks you very much
